applications/TrilinosApplication/tests/cpp_tests/test_trilinos_space.cpp

```diff
@@ -0,0 +1,145 @@
+//  KRATOS  _____     _ _ _
+//         |_   _| __(_) (_)_ __   ___  ___
+//           | || '__| | | | '_ \ / _ \/ __|
+//           | || |  | | | | | | | (_) \__
+//           |_||_|  |_|_|_|_| |_|\___/|___/ APPLICATION
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "testing/testing.h"
+#include "trilinos_space.h"
+#include "containers/model.h"
+#include "mpi/includes/mpi_data_communicator.h"
+
+namespace Kratos::Testing
+{
+
+/// The sparse matrix type
+using TrilinosSparseSpaceType = TrilinosSpace<Epetra_FECrsMatrix, Epetra_FEVector>;
+using SparseMatrixType = TrilinosSparseSpaceType::MatrixType;
+using VectorType = TrilinosSparseSpaceType::VectorType;
+
+/**
+ * @brief Generates a dummy diagonal sparse matrix for Trilinos
+ * @param rDataCommunicator The data communicator considered
+ * @param NumGlobalElements The global dimension of the matrix
+ * @param Offset The offset considered
+ */
+SparseMatrixType GenerateDummySparseMatrix(
+    const DataCommunicator& rDataCommunicator,
+    const int NumGlobalElements = 12,
+    const double Offset = 0.0
+    )
+{
+    // Generate Epetra communicator
+    KRATOS_ERROR_IF_NOT(rDataCommunicator.IsDistributed()) << "Only distributed DataCommunicators can be used!" << std::endl;
+    auto raw_mpi_comm = MPIDataCommunicator::GetMPICommunicator(rDataCommunicator);
+    Epetra_MpiComm epetra_comm(raw_mpi_comm);
+
+    // Create a map
+    Epetra_Map Map(NumGlobalElements,0,epetra_comm);
+
+    // Local number of rows
+    const int NumMyElements = Map.NumMyElements();
+
+    // Get update list
+    int* MyGlobalElements = Map.MyGlobalElements( );
+
+    // Create an integer vector NumNz that is used to build the EPetra Matrix.
+    // NumNz[i] is the Number of OFF-DIAGONAL term for the ith global equation
+    // on this processor
+    int* NumNz = new int[NumMyElements];
+    for ( int i=0; i<NumMyElements; i++) {
+        NumNz[i] = 1;
+    }
+
+    // Create a Epetra_Matrix
+    SparseMatrixType A(Copy, Map, NumNz);
```

Comment:
```suggestion
    std::vector<int> NumNz(NumMyElements, 1);
    
    // Create a Epetra_Matrix
    SparseMatrixType A(Copy, Map, NumNz.data());
```

no need to work with manual memory allocation (which BTW you forgot to `delete`)